postgres-types/src/lib.rs

```diff
@@ -391,6 +397,10 @@ impl WrongType {
 /// | `chrono::DateTime<FixedOffset>` | TIMESTAMP WITH TIME ZONE            |
 /// | `chrono::NaiveDate`             | DATE                                |
 /// | `chrono::NaiveTime`             | TIME                                |
+/// | `time::PrimitiveDateTime`       | TIMESTAMP                           |
+/// | `time::OffsetDateTime`          | TIMESTAMP, TIMESTAMP WITH TIME ZONE |
```

Comment:
I think this should only support TIMESTAMP WITH TIME ZONE to align with chrono.

Reply:
Chrono's types embed specific timezones (or rather `UTC` vs `Local`) where as `OffsetDateTime` doesn't have the generic component and is compatible with both `TIMESTAMP` and `TIMESTAMPZ`. I'd prefer to be able to use `OffsetDateTime` with `TIMESTAMP`.

Reply:
> I'd prefer to be able to use OffsetDateTime with TIMESTAMP.

With [cockroach](https://www.cockroachlabs.com/docs/v19.2/timestamp.html#variants) `TIMESTAMP` will always be `UTC`. I'm still looking for clarification on postgres's behavior. I think with the fact that we're explicitly setting the session to `UTC`, it should be OK either way.

Reply:
@sfackler I thought about this some more and came to the same conclusion. `OffsetDateTime` now only supports `TIMESTAMPTZ` in the latest commit. Do you want me to squash these commits?